src/features/change-language/ui/ChangeLanguageSelector.tsx

```diff
@@ -0,0 +1,52 @@
+import { FC } from 'react';
+
+import { useNavigation } from '@react-navigation/native';
+import { useTranslation } from 'react-i18next';
+
+import { colors } from '@shared/lib';
+import {
+  YStack,
+  XStack,
+  Box,
+  BoxProps,
+  Text,
+  ArrowRightIcon,
+} from '@shared/ui';
+
+const ChangeLanguageSelector: FC<BoxProps> = props => {
+  const { t, i18n } = useTranslation();
+  const navigation = useNavigation();
+  const languagesAvailable = Object.keys(i18n.services.resourceStore.data);
+  const { resolvedLanguage } = i18n;
+
+  const changeLanguage = async (locale: string) => {
```

Comment:
As this function is a part of business logic and doesn't depend on react render then please move it to "model" segment.
The function here should look like this:
```
const changeLanguage = (locale: string) => {
  changeLanguageModel.changeLanguage(locale);
  onChange();
}
```